### PR TITLE
[NETBEANS-3107] - cleanup use of generics

### PR DESCRIPTION
**Pull request by BradWalker**
I'm trying to clean up the use of generics when there is a private method or variable.

For example, we see a lot of the follow code in the source tree:

`private Map/*<Connection, Driver>*/ conn2Driver = new WeakHashMap();`

Change this to

`private Map<Connection, Driver> conn2Driver = new WeakHashMap<>();`

Java really wants us to use Generics and this looks like it was a work in progress. So I'm simply
trying to complete some of it.

### CODE REVIEW

File: profiler/lib.profiler/src/org/netbeans/lib/profiler/results/cpu/cct/CCTResultsFilter.java [hunk 1]
```diff
@@ -46,7 +46,7 @@
     }
     
     public static interface EvaluatorProvider {
-        Set/*<Evaluator>*/ getEvaluators();
+        Set<Evaluator> getEvaluators();
```
Comment (matthiasblaesing) [inline profiler/lib.profiler/src/org/netbeans/lib/profiler/results/cpu/cct/CCTResultsFilter.java]: This touches the interface of the module, there are more places in the same code, that could get some generic love, but that is not "cleanup", so could you please revert this change?

File: profiler/lib.profiler/src/org/netbeans/lib/profiler/results/BaseCallGraphBuilder.java [hunk 2]
```diff
@@ -42,7 +42,7 @@
 
     //~ Instance fields ----------------------------------------------------------------------------------------------------------
 
-    protected List /*<Runnable>*/ afterBatchCommands = new ArrayList /*<Runnable>*/();
+    protected List <Runnable> afterBatchCommands = new ArrayList<>();
```
Comment (matthiasblaesing) [inline profiler/lib.profiler/src/org/netbeans/lib/profiler/results/BaseCallGraphBuilder.java]: See comment in `CCTResultsFilter.java`.

Comment (BradWalker) [reply profiler/lib.profiler/src/org/netbeans/lib/profiler/results/BaseCallGraphBuilder.java]: I should have thought about that.. But, when I start making these mass changes, I kinda get into a rhythm.

And that's why we do code reviews..

Thanks for the code review!

**Comment by sdedic** [comment]
Any objections to merge ?

**Comment by neilcsmith-net** [comment]
@sdedic yes, big objection right now! We're in merge freeze for 11.2. It will be announced on dev@ when master is reopened.

**Comment by matthiasblaesing** [comment]
Merged via https://github.com/apache/netbeans/commit/f2611e358c181935500ea4d9d9142fb850504a72. Thank you.
